javascript/packages/linter/test/cli.test.ts

```diff
@@ -336,6 +336,110 @@ describe("CLI Output Formatting", () => {
     })
   })
 
+  describe("--fail-on-warning", () => {
+    const { writeFileSync, unlinkSync } = require("fs")
+    const configPath = "test/fixtures/.herb.yml"
+
+    test("exits with error code when warnings are present with --fail-on-warning flag", () => {
+      try {
+        writeFileSync(configPath, dedent`
+          linter:
+            rules:
+              html-img-require-alt:
+                severity: warning
+              html-tag-name-lowercase:
+                enabled: false
+        `)
+
+        const { output, exitCode } = runLinter("test-file-with-errors.html.erb", "--fail-on-warning")
+
+        expect(output).toContain("warning")
+        expect(exitCode).toBe(1)
+      } finally {
+        try { unlinkSync(configPath) } catch {}
+      }
+    })
+
+    test("exits with success when no warnings are present with --fail-on-warning flag", () => {
+      try {
+        writeFileSync(configPath, dedent`
+          linter:
+            rules:
+              html-img-require-alt:
+                severity: warning
+        `)
+
+        const { output, exitCode } = runLinter("clean-file.html.erb", "--fail-on-warning")
+
+        expect(exitCode).toBe(0)
+      } finally {
+        try { unlinkSync(configPath) } catch {}
+      }
+    })
+
+    test("exits with error code when failOnWarning is set in config", () => {
+      try {
+        writeFileSync(configPath, dedent`
+          linter:
+            failOnWarning: true
```

Comment:
I'm wondering if we should use kabab-case here for the key: `fail-on-warning`? 

Or maybe it would make more sense to use a more flexible key, like: `fail-level: warning`. That way we could even say `fail-level: info`

Reply:
I tried using flexible key. What do you think?